### PR TITLE
Support URLEncoded client cert

### PR DESCRIPTION
**Pull request by pmd1nh**

### CODE REVIEW

**Comment by pmd1nh** [comment]
#build       #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_part2,com.ibm.ws.webcontainer_fat_servlet31toleration,com.ibm.ws.webcontainer_fat_servlet31toleration_part2,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0_part2,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by pmd1nh** [comment]
https://github.com/OpenLiberty/open-liberty/issues/11680

**Comment by pmd1nh** [comment]
com.ibm.ws.fat.wc.tests.WCTestEncodedX590 | 6 | 0 | 0 | 9 s | 23
-- | -- | -- | -- | -- | --
test_Good_EncodedX590ClientCert | 1 | 0 | 0 | 4 s | 1
test_Bad_EncodedX590ClientCert | 1 | 0 | 0 | 0 s | 2
test_None_EncodedX590ClientCert | 1 | 0 | 0 | 0 s | 3
test_Good_EncodedX590ClientCert_EE9_FEATURES | 1 | 0 | 0 | 4 s | 4
test_Bad_EncodedX590ClientCert_EE9_FEATURES | 1 | 0 | 0 | 0 s | 5
test_None_EncodedX590ClientCert_EE9_FEATURES | 1 | 0 | 0 | 0 s | 6

**Comment by pmd1nh** [comment]
#run-libby-bot

File: dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCTestEncodedX590.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCTestEncodedX590.java:3]: If this is a new test class there is no need for a 2020 copyright date.

File: dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCTestEncodedX590.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+import componenttest.annotation.ExpectedFFDC;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyServer;
+
+@RunWith(FATRunner.class)
+@Mode(TestMode.FULL)
+public class WCTestEncodedX590 {
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCTestEncodedX590.java:45]: Can you add a high-level summary of what we're testing here as a comment?

File: dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCTestEncodedX590.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+        // Start the server and use the class name so we can find logs easily.
+        server.startServer(WCTestEncodedX590.class.getSimpleName() + ".log");
+    }
+
+    @AfterClass
+    public static void testCleanup() throws Exception {
+        if (server != null && server.isStarted()) {
+            server.stopServer("SRVE8086E");
+        }
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCTestEncodedX590.java:71]: Which test causes this error? Would you mind adding a comment that shows what the Error text is so we a bit more clear?

File: dev/com.ibm.ws.webcontainer.servlet.4.0_fat/publish/servers/servlet40_X590CertServer/bootstrap.properties [hunk 4]
```diff
@@ -0,0 +1,17 @@
+###############################################################################
+# Copyright (c) 2017, 2021 IBM Corporation and others.
+# All rights reserved. This program and the accompanying materials
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.4.0_fat/publish/servers/servlet40_X590CertServer/bootstrap.properties:3]: new file no need for 2017 copyright date.

File: dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCTestEncodedX590.java [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+                LOG.info("\nResponse content: " + content);
+
+                assertTrue("Response did not contain expected response: ", content.contains(expectedResponse));
+            }
+        }
+
+        LOG.info("\n #################### [test_None_EncodedX590ClientCert] FINISH #################### ");
+    }
+
+}
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCTestEncodedX590.java]: newline please.

File: dev/com.ibm.ws.webcontainer.servlet.4.0_fat/publish/servers/servlet40_X590CertServer/.gitignore [hunk 6]
```diff
@@ -0,0 +1,2 @@
+/apps
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.4.0_fat/publish/servers/servlet40_X590CertServer/.gitignore]: I don't think we need the `/apps` here looking at your test class you're only using dropins.

File: dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/request/IRequestImpl.java [hunk 7]
```diff
@@ -354,19 +354,45 @@ public String getMethod()
    */
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/request/IRequestImpl.java:355]: Update the copyright header for IRequestImpl.java to include 2021

Comment (pnicolucci) [reply dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/request/IRequestImpl.java:355]: We also need a way to beta guard this functionality.

**Comment by pmd1nh** [comment]
beta guarded

**Comment by pmd1nh** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_part2,com.ibm.ws.webcontainer_fat_servlet31toleration,com.ibm.ws.webcontainer_fat_servlet31toleration_part2,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0_part2,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by pmd1nh** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_part2,com.ibm.ws.webcontainer_fat_servlet31toleration,com.ibm.ws.webcontainer_fat_servlet31toleration_part2,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0_part2,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Review by pnicolucci — APPROVED** [review]
This looks good to me Phu. I pulled your changes in locally and tested with and without the jvm.properties and things worked as we would expect them to.

Pass with -Dcom.ibm.ws.beta.edition=true:
test_Good_EncodedX590ClientCert
test_Bad_EncodedX590ClientCert
test_None_EncodedX590ClientCert

Fail with -Dcom.ibm.ws.beta.edition=true removed: 
test_Good_EncodedX590ClientCert

**Comment by pmd1nh** [comment]
yep, the test_Good_EncodedX590ClientCert has new encoding format so it should fail in non-beta.

**Comment by pmd1nh** [comment]
#run-libby-bot

**Pull request merged**